Userland/Libraries/LibJS/Runtime/Intl/DurationFormat.cpp

```diff
@@ -183,7 +183,62 @@ ThrowCompletionOr<Temporal::DurationRecord> to_duration_record(GlobalObject& glo
     return result;
 }
 
-// 1.1.2 GetDurationUnitOptions ( unit, options, baseStyle, stylesList, digitalBase, prevStyle ), https://tc39.es/proposal-intl-duration-format/#sec-getdurationunitoptions
+// 1.1.4 DurationSign ( record ), https://tc39.es/proposal-intl-duration-format/#sec-durationsign
+i8 duration_sign(Temporal::DurationRecord const& record)
+{
+    // 1. For each row in Table 1, except the header row, in table order, do
+    for (auto const& duration_instances_component : duration_instances_components) {
+        // a. Let valueSlot be the Value Slot value.
+        auto value_slot = duration_instances_component.value_slot;
+
+        // b. Let v be value of the valueSlot slot of record.
+        auto value = record.*value_slot;
+
+        // c. If v < 0, return -1.
+        if (value < 0)
+            return -1;
+
+        // d. If v > 0, return 1.
+        if (value > 0)
+            return 1;
+    }
+
+    // 2. Return 0.
+    return 0;
+}
+
+// 1.1.5 IsValidDurationRecord ( record ), https://tc39.es/proposal-intl-duration-format/#sec-isvaliddurationrecord
+bool is_valid_duration_record(Temporal::DurationRecord const& record)
+{
+    // 1. Let sign be ! DurationSign(record).
```

Comment:
Ah, they'll eventually have to change that to the new non-! syntax for AOs not returning a completion